src/earn/EarnActivePool.tsx

```diff
@@ -33,26 +36,33 @@ function PoolDetailsLoading() {
   )
 }
 
-export default function EarnActivePool() {
+interface Props {
+  buttonDisplay: 'ExitAndDeposit' | 'ViewPools'
+  depositTokenId: string
+  poolTokenId: string
+}
+
+export default function EarnActivePool({ depositTokenId, poolTokenId, buttonDisplay }: Props) {
   const { t } = useTranslation()
-  const token = useTokenInfo(networkConfig.arbUsdcTokenId)
-  const poolToken = useTokenInfo(networkConfig.aaveArbUsdcTokenId)
+  const depositToken = useTokenInfo(depositTokenId)
+  const poolToken = useTokenInfo(poolTokenId)
   const asyncPoolInfo = useAsync(async () => {
-    if (!token || !token.address) {
+    if (!depositToken || !depositToken.address) {
       // should never happen
-      Logger.warn(TAG, `Token with id ${networkConfig.arbUsdcTokenId} not found`)
-      throw new Error(`Token with id ${networkConfig.arbUsdcTokenId} not found`)
+      Logger.warn(TAG, `Token with id ${depositTokenId} not found`)
+      throw new Error(`Token with id ${depositTokenId} not found`)
     }
-    if (!isAddress(token.address)) {
-      Logger.warn(
-        TAG,
-        `Token with id ${networkConfig.arbUsdcTokenId} does not contain a valid address`
-      )
-      throw new Error(
-        `Token with id ${networkConfig.arbUsdcTokenId} does not contain a valid address`
-      )
+
+    if (!isAddress(depositToken.address)) {
+      Logger.warn(TAG, `Token with id ${depositTokenId} does not contain a valid address`)
+      throw new Error(`Token with id ${depositTokenId} does not contain a valid address`)
     }
-    return fetchAavePoolInfo(token.address)
+
+    return fetchAavePoolInfo({
+      assetAddress: depositToken.address,
+      contractAddress: networkConfig.arbAavePoolV3ContractAddress,
+      network: Network.Arbitrum,
```

Comment:
this can be inferred from the `depositToken`

Reply:
Added in f9cc578ea68017534f90b1af611f114103bcdb60. I wonder if all the type guards for `fetchAavePoolInfo` are a bit overkill instead of typecasting and using optional chaining having  `fetchAavePoolInfo` throw instead. Now whenever someone wants to use `fetchAavePoolInfo` they'll need to use about ~25 lines of type guards to use this function. It might make sense to make these guards reusable the next time we use `fetchAavePoolInfo` somewhere.

Reply:
Made these top checks a bit shorter in 27c08ed0bda6c33e21d96a71a7216452d9264f39. It might be worthwhile to create a `isValidToken` util function to cover the remaining checks if we are finding we are reusing them frequently.